controllers/rollingupgrade_controller.go

```diff
@@ -701,10 +701,11 @@ func (r *RollingUpgradeReconciler) Process(ctx *context.Context,
 	nodesProcessed, err := r.runRestack(ctx, ruObj, KubeCtlBinary)
 	if err != nil {
 		r.error(ruObj, err, "Failed to runRestack")
-		return r.finishExecution(StatusError, nodesProcessed, ctx, ruObj)
+		r.finishExecution(StatusError, nodesProcessed, ctx, ruObj)
+		return
 	}
 
-	return r.finishExecution(StatusComplete, nodesProcessed, ctx, ruObj)
+	r.finishExecution(StatusComplete, nodesProcessed, ctx, ruObj)
```

Comment:
Does it make more sense to defer this call

Reply:
Good idea. Done.

Reply:
Actually.. maybe I spoke too soon. There is at least one case where we don't want to call finishExecution; when the CR is already in completed state. To get past this, we could do `defer finishExecution` after the check. But then there are other parameters needed for finishExecution that need to be set correctly: status, nodesProcessd. This makes the code complex and hard to read. One has to remember to set these variables to the correct value before the function ends.

Let's keep this as is for now. There is a need to refactor a bunch of this code separately.